core/src/main/scala/chisel3/properties/Property.scala

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.properties
+
+import chisel3.{ActualDirection, IsBindable, SpecifiedDirection}
+import chisel3.internal.Binding
+import scala.reflect.runtime.universe.{typeOf, TypeTag}
+import scala.annotation.implicitNotFound
+
+/** PropertyType defines a typeclass for valid Property types.
+  *
+  * Typeclass instances will be defined for Scala types that can be used as
+  * properties. This includes builtin Scala types as well as types defined in
+  * Chisel.
+  */
+@implicitNotFound("unsupported Property type ${T}")
+private[chisel3] class PropertyType[T]
+
+/** Companion object for PropertyType.
+  *
+  * Typeclass instances for valid Property types are defined here, so they will
+  * be in the implicit scope and available for users.
+  */
+private[chisel3] object PropertyType {}
```

Comment:
I don't know that these can really be package private and yet used in a public interface (on `Property` below), I think we will need to make these public.

Reply:
In a later PR, where I actually add some typeclass instances here, it seems to work fine from the unit tests. Maybe I'm missing something--is there something I should test that would require this object to be public?

Reply:
Perhaps it's changed in the last few days, but I don't see any unit tests calling this, only 1 that checks that it fails to compile. I think if you actually try to use this code from outside of the `chisel3` package, you will get compile errors.

Reply:
This draft PR on my fork is no longer in sync, but I think it indicates what I mean: https://github.com/mikeurbach/chisel3/pull/1/files#diff-4b64ebc257d840c404c27e41af276a1ae760940bdf80acf7f0cac225100fc944R20

There are tests in the chiselTests package that successfully create properties, and PropertyType is still package private. Maybe I'm missing something, but it seems like just `Property` needs to be public, and Scala implicit resolution did the rest.